### PR TITLE
avoid duplicate error event

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:

this is from code read when doing other PR 

seems we have event https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/pkg/cloud/services/networking/trunk.go#L73 so no need duplicate event to confuse admin

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**TODOs**:

- [ ] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

### CODE REVIEW

**Comment by jichenjc** [comment]
there's only one usage, so update directly in the function...

File: pkg/cloud/services/networking/port.go [hunk 1]
```diff
@@ -179,7 +179,6 @@ func (s *Service) GetOrCreatePort(eventObject runtime.Object, clusterName string
 	if portOpts.Trunk != nil && *portOpts.Trunk {
 		trunk, err := s.getOrCreateTrunk(eventObject, clusterName, port.Name, port.ID)
 		if err != nil {
-			record.Warnf(eventObject, "FailedCreateTrunk", "Failed to create trunk for port %s: %v", portName, err)
```
Comment (mdbooth) [inline pkg/cloud/services/networking/port.go:182]: I was thinking keep this one and remove the events in getOrCreateTrunk()? By emitting the event at both error points we're no longer avoiding duplicating the error event.

Comment (dulek) [reply pkg/cloud/services/networking/port.go:182]: Yeah, I'd probably go with what Matt says here, but no strong preference.

**Pull request merged**